b0eb2c372597de47b8df23a6421e6176a39393ed
src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java
--- 
+++ 
@@ -258,8 +258,14 @@
     @Override
     public Integer call() throws Exception {
       Injector injector = Guice.createInjector(new LuceneServerModule(this));
-      LuceneServer luceneServer = injector.getInstance(LuceneServer.class);
-      luceneServer.start();
+      LuceneServer luceneServer;
+      try {
+        luceneServer = injector.getInstance(LuceneServer.class);
+        luceneServer.start();
+      } catch (Throwable t) {
+        logger.error("Uncaught exception", t);
+        throw t;
+      }
       luceneServer.blockUntilShutdown();
       return 0;
     }